datastore.go

```diff
@@ -79,11 +83,17 @@ func NewDatastore(path string, options *Options) (*Datastore, error) {
 // can be mutated without incurring changes to the underlying Datastore until
 // the transaction is Committed.
 func (d *Datastore) NewTransaction(readOnly bool) ds.Txn {
-	return &txn{d.DB.NewTransaction(!readOnly)}
+	return &txn{d.DB.NewTransaction(!readOnly), false}
```

Comment:
now that we have more than one value in the struct, let's address them by name

Reply:
Oops, I merged before addressing this. Worth opening another PR?

Reply:
maybe just leave a todo for yourself! i think it's fine.